src/frontend/js/components/DashBoard/ProtectedRoute.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { ReactNode } from 'react';
+import { Navigate, Outlet } from 'react-router-dom';
+
+interface ProtectedRouteProps {
+  isAllowed: Boolean;
+  redirectPath: string;
+  children: ReactNode;
+}
+
+const ProtectedRoute = ({ isAllowed, redirectPath, children }: ProtectedRouteProps) => {
+  if (!isAllowed) {
+    return <Navigate to={redirectPath} replace />;
+  }
+  return children || <Outlet />;
```

Comment:
Hmmm... If we need to use an outlet, I think it's better to put it as children during the route instantiation?
```suggestion
  return children;
```